### ISSUE TITLE
mklive: lightdm no longer works

### ISSUE THREAD

**Issue by absolutelynothinghere** [void-linux/void-mklive#371]
I updated `mklive` today and found an issue that was not present in older versions (last clone c. 2022).

The lightdm greeter is no longer displayed, instead I see the "regular" tty login prompt. If I login through that and manually `startx` I reach the desktop (XFCE) but an error is displayed saying "[XFCE PolicyKit Agent](https://github.com/void-linux/void-packages/issues/20245)" and Whisker Menu doesn't work.

**Comment by classabbyamp** [comment]
I cannot reproduce this with neither an x86_64 XFCE image generated on [`20240217`](https://devspace.voidlinux.org/abby/live/void-live-20240217/) nor an image generated just now, in a VM. It showed me the TTY login for a second, then a black screen while lightdm loaded, then it auto-logged into XFCE. Logging out goes to lightdm just fine.

what exact command did you use to generate the image? what hardware is the running on? (especially interested in the graphics card)

**Comment by classabbyamp** [comment]
the service isn't enabled automatically, you need to do that with `-S`

**Comment by classabbyamp** [comment]
lightdm requires dbus, did you enable that?

BTW, by using `mklive` directly, you're missing [some of the things](https://github.com/void-linux/void-mklive/blob/master/build-x86-images.sh#L92-L95) that the xfce images have, because `build_x86_images` does them

**Comment by classabbyamp** [comment]
> Quick question; if I add `-C "live.autologin"` do I still need the lightdm package and service?

yes, because lightdm starts xfce. if you just had `live.autologin`, you'd end up at tty1 with a shell prompt

**Comment by classabbyamp** [comment]
did you put `xfce` in `/etc/lightdm/.session` in an include dir?

https://github.com/void-linux/void-mklive/blob/master/dracut%2Fvmklive%2Fdisplay-manager-autologin.sh#L33-L41

**Comment by classabbyamp** [comment]
look at what `build-x86-images` is doing. all you need to do is have an include dir with `/etc/lightdm/.session` containing `xfce`, and have `live.autologin` on the commandline. the rest is handled by dracut on boot.
